include/morphio/morphology.h

```diff
@@ -99,17 +100,17 @@ class Morphology
 
        If id == -1, the iteration will start at each root section, successively
     **/
-    depth_iterator depth_begin() const;
-    depth_iterator depth_end() const;
+    depth_iterator_t<Section, Morphology> depth_begin() const;
+    depth_iterator_t<Section, Morphology> depth_end() const;
 
     /**
        Breadth first iterator
 
        If id == -1, the iteration will be successively performed starting
        at each root section
     **/
-    breadth_iterator breadth_begin() const;
-    breadth_iterator breadth_end() const;
+    breadth_iterator_t<Section, Morphology> breadth_begin() const;
+    breadth_iterator_t<Section, Morphology> breadth_end() const;
 
     /**
      * Return the soma type
```

Comment:
Can you nuke the commented function `traverse` a bit below in this file ? Thanks